src/com/nutrons/framework/util/FlowOperators.java

```diff
@@ -70,4 +71,22 @@
   public static <T> T getLastValue(Flowable<T> input) {
     return input.blockingLatest().iterator().next();
   }
+
+  public static FlowableTransformer<Double, Double> pidLoop(double proportional,
+                                                            int integralBuffer,
+                                                            double integral,
+                                                            double derivative) {
+    return error -> {
+        Flowable<Double> errorP = error.map(x -> x * proportional);
+        Flowable<Double> errorI = error.buffer(integralBuffer, 1)
+            .map(list -> list.stream().reduce(0.0, (x, acc) -> x + acc))
```

Comment:
This is probably called frequently enough that you want an efficient implementation with a queue and a scan.